src/dataflow-types/src/reconciliation/command.rs

```diff
@@ -35,9 +40,45 @@ use tracing::warn;
 
 use mz_repr::GlobalId;
 
-use crate::client::{ComputeClient, ComputeCommand, ComputeResponse, GenericClient};
+use crate::client::{
+    ComputeClient, ComputeCommand, ComputeResponse, GenericClient, StorageClient, StorageCommand,
+    StorageResponse, TimestampBindingFeedback,
+};
+use crate::sources::SourceConnector;
 use crate::{DataflowDescription, Plan};
 
+/// Generic trait to reconcile commands targeted at a COMPUTE or STORAGE instance.
+#[async_trait]
+pub trait CommandReconcile<C, R>: fmt::Debug + Send
```

Comment:
I like the intention, but I'm not sure this is in the spirit we would like to evolve storage and compute. In principle, the storage command reconciliation should live in the storage crate, as it doesn't exist in the controller. Extracting a common trait makes this more complicated. Maybe @petrosagg has some opinion on this.

Reply:
Related: https://github.com/MaterializeInc/materialize/issues/12211

Reply:
Yeah that makes sense, I reverted the commit.